### ISSUE TITLE
Multi-doc YAML from URL

### ISSUE THREAD

**Issue by errordeveloper** [kubernetes/kubernetes#21764]
Looks like when deploying a resource from a URL multi-doc YAML is not handled properly, unlike with a local file.

This results in  only first document being created (i.e. the replication controller):

```
kubectl create -f https://raw.githubusercontent.com/kubernetes/dashboard/e22812c45bd8c8b2afb135e73c4f151a178de6a8/src/deploy/kubernetes-dashboard-canary.yaml
```

This creates both, the service and the replication controller:
```
curl -sLO https://raw.githubusercontent.com/kubernetes/dashboard/e22812c45bd8c8b2afb135e73c4f151a178de6a8/src/deploy/kubernetes-dashboard-canary.yaml
kubectl create -f kubernetes-dashboard-canary.yaml
```

However, using a URL and `kind: List` works fine:
```
kubectl create -f https://raw.githubusercontent.com/kubernetes/dashboard/0e5b5ecadc3e2959e0d17b3dec2ea111a6540279/src/deploy/kubernetes-dashboard-canary.yaml
```

The [exact difference between two files](https://github.com/kubernetes/dashboard/commit/fb559a5740ec0dc4a30a026b2ecd1a805d941608) is basically:

```diff
diff --git a/src/deploy/kubernetes-dashboard-canary.yaml b/src/deploy/kubernetes-dashboard-canary.yaml
index ce62378..f719ab5 100644
--- a/src/deploy/kubernetes-dashboard-canary.yaml
+++ b/src/deploy/kubernetes-dashboard-canary.yaml
@@ -1,46 +1,49 @@
+kind: List
 apiVersion: v1
-kind: ReplicationController
-metadata:
...
----
-apiVersion: v1
-kind: Service
...
+items:
+- kind: ReplicationController
+  apiVersion: v1
...
+- kind: Service
+  apiVersion: v1
...
```
namely
```
kind: List
items:
- kind: ReplicationController
- kind: Service
```
_VS_
```
kind: ReplicationController
---
kind: Service
```
This was tested on v1.1.7.

**Comment by errordeveloper** [comment]
Would be make sense to try verifying if this will work instead:
```
---
kind: ReplicationController
---
kind: Service
```

**Comment by jeffmendoza** [comment]
Fixed by #22009
